### PR TITLE
Blogv3

### LINKED ISSUE #8: Blog v3
Creation of back office: administration of blog posts, comments and user for the admin

### LINKED ISSUE #9: Add register user and forgot password

### LINKED ISSUE #10: Update readme

### LINKED ISSUE #12: add user picture

### LINKED ISSUE #14: add comment approval

### PR DESCRIPTION
**Pull request by Mamoon97150**
closes #8

### CODE REVIEW

File: public/js/extra/chart-pie.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Set new default font family and font color to mimic Bootstrap's default styling
+Chart.defaults.global.defaultFontFamily = 'Nunito', '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif';
+Chart.defaults.global.defaultFontColor = '#858796';
```
Comment (Mamoon97150) [inline public/js/extra/chart-pie.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
Chart.defaults.global.defaultFontColor = "#858796";
```

File: public/js/extra/chart-area.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    toFixedFix = function(n, prec) {
+      var k = Math.pow(10, prec);
+      return '' + Math.round(n * k) / k;
+    };
+  // Fix for IE parseFloat(0.55).toFixed(0) = 0;
+  s = (prec ? toFixedFix(n, prec) : '' + Math.round(n)).split('.');
+  if (s[0].length > 3) {
+    s[0] = s[0].replace(/\B(?=(?:\d{3})+(?!\d))/g, sep);
+  }
+  if ((s[1] || '').length < prec) {
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:23]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
  if ((s[1] || "").length < prec) {
```

File: public/js/extra/chart-area.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            xPadding: 15,
+            yPadding: 15,
+            displayColors: false,
+            intersect: false,
+            mode: 'index',
+            caretPadding: 10,
+            callbacks: {
+              label: function(tooltipItem, chart) {
+                var datasetLabel = chart.datasets[tooltipItem.datasetIndex].label || '';
+                return datasetLabel + ': $' + number_format(tooltipItem.yLabel);
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:132]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
                return datasetLabel + ": $" + number_format(tooltipItem.yLabel);
```

File: public/js/extra/chart-area.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+Chart.defaults.global.defaultFontFamily = 'Nunito', '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif';
+Chart.defaults.global.defaultFontColor = '#858796';
+
+function number_format(number, decimals, dec_point, thousands_sep) {
+  // *     example: number_format(1234.56, 2, ',', ' ');
+  // *     return: '1 234,56'
+  number = (number + '').replace(',', '').replace(' ', '');
+  var n = !isFinite(+number) ? 0 : +number,
+    prec = !isFinite(+decimals) ? 0 : Math.abs(decimals),
+    sep = (typeof thousands_sep === 'undefined') ? ',' : thousands_sep,
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
    sep = (typeof thousands_sep === "undefined") ? ',' : thousands_sep,
```

File: public/js/extra/datatables.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// Call the dataTables jQuery plugin
+$(document).ready(function() {
+  $('#postsTable').DataTable();
```
Comment (Mamoon97150) [inline public/js/extra/datatables.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
  $("#postsTable").DataTable();
```

File: public/js/extra/chart-area.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    dec = (typeof dec_point === 'undefined') ? '.' : dec_point,
+    s = '',
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:13]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
    s = "",
```

File: public/js/extra/chart-area.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                maxTicksLimit: 7
+              }
+            }],
+            yAxes: [{
+              ticks: {
+                maxTicksLimit: 5,
+                padding: 10,
+                // Include a dollar sign in the ticks
+                callback: function(value, index, values) {
+                  return '' + number_format(value);
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:100]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
                  return "" + number_format(value);
```

File: public/js/extra/chart-pie.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        percent.push(dataPie[i].posts_count);
+      }
+
+      var chartdata = {
+        labels: name,
+        datasets: [
+          {
+            label: "Posts",
+            data: percent,
+            backgroundColor: ['#ffb032', '#4e73df', '#1cc88a', '#36b9cc'],
```
Comment (Mamoon97150) [inline public/js/extra/chart-pie.js:25]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
            backgroundColor: ["#ffb032", '#4e73df', '#1cc88a', '#36b9cc'],
```

File: public/js/extra/chart-area.js [hunk 9]
```diff
@@ -0,0 +1,144 @@
+// Set new default font family and font color to mimic Bootstrap's default styling
+Chart.defaults.global.defaultFontFamily = 'Nunito', '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif';
+Chart.defaults.global.defaultFontColor = '#858796';
+
+function number_format(number, decimals, dec_point, thousands_sep) {
+  // *     example: number_format(1234.56, 2, ',', ' ');
+  // *     return: '1 234,56'
+  number = (number + '').replace(',', '').replace(' ', '');
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
  number = (number + "").replace(',', '').replace(' ', '');
```

File: public/js/extra/chart-pie.js (hunk 8 above)
Comment (Mamoon97150) [inline public/js/extra/chart-pie.js:25]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
            backgroundColor: ["#ffb032", '#4e73df', '#1cc88a', '#36b9cc'],
```

File: public/js/extra/chart-pie.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+            hoverBackgroundColor: ['#ff9d00', '#2e59d9', '#17a673', '#2c9faf'],
+            hoverBorderColor: "rgba(234, 236, 244, 1)",
+          }
+        ]
+      };
+
+      var ctx = $("#myPieChart");
+
+      var LineGraph = new Chart(ctx, {
+        type: 'doughnut',
```
Comment (Mamoon97150) [inline public/js/extra/chart-pie.js:35]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
        type: "doughnut",
```

File: public/js/extra/chart-area.js (hunk 3 above)
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:132]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
                return datasetLabel + ": $" + number_format(tooltipItem.yLabel);
```

File: public/js/extra/chart-area.js [hunk 11]
```diff
@@ -0,0 +1,144 @@
+// Set new default font family and font color to mimic Bootstrap's default styling
+Chart.defaults.global.defaultFontFamily = 'Nunito', '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif';
+Chart.defaults.global.defaultFontColor = '#858796';
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
Chart.defaults.global.defaultFontColor = "#858796";
```

File: public/js/extra/chart-area.js [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+      return '' + Math.round(n * k) / k;
+    };
+  // Fix for IE parseFloat(0.55).toFixed(0) = 0;
+  s = (prec ? toFixedFix(n, prec) : '' + Math.round(n)).split('.');
+  if (s[0].length > 3) {
+    s[0] = s[0].replace(/\B(?=(?:\d{3})+(?!\d))/g, sep);
+  }
+  if ((s[1] || '').length < prec) {
+    s[1] = s[1] || '';
+    s[1] += new Array(prec - s[1].length + 1).join('0');
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:25]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
    s[1] += new Array(prec - s[1].length + 1).join("0");
```

File: public/js/extra/chart-area.js [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+    prec = !isFinite(+decimals) ? 0 : Math.abs(decimals),
+    sep = (typeof thousands_sep === 'undefined') ? ',' : thousands_sep,
+    dec = (typeof dec_point === 'undefined') ? '.' : dec_point,
+    s = '',
+    toFixedFix = function(n, prec) {
+      var k = Math.pow(10, prec);
+      return '' + Math.round(n * k) / k;
+    };
+  // Fix for IE parseFloat(0.55).toFixed(0) = 0;
+  s = (prec ? toFixedFix(n, prec) : '' + Math.round(n)).split('.');
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:19]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
  s = (prec ? toFixedFix(n, prec) : "" + Math.round(n)).split('.');
```

File: public/js/extra/chart-area.js [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+                maxTicksLimit: 7
+              }
+            }],
+            yAxes: [{
+              ticks: {
+                maxTicksLimit: 5,
+                padding: 10,
+                // Include a dollar sign in the ticks
+                callback: function(value, index, values) {
+                  return '' + number_format(value);
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:100]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
                  return "" + number_format(value);
```

File: public/js/extra/chart-area.js [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+    prec = !isFinite(+decimals) ? 0 : Math.abs(decimals),
+    sep = (typeof thousands_sep === 'undefined') ? ',' : thousands_sep,
+    dec = (typeof dec_point === 'undefined') ? '.' : dec_point,
+    s = '',
+    toFixedFix = function(n, prec) {
+      var k = Math.pow(10, prec);
+      return '' + Math.round(n * k) / k;
+    };
+  // Fix for IE parseFloat(0.55).toFixed(0) = 0;
+  s = (prec ? toFixedFix(n, prec) : '' + Math.round(n)).split('.');
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:19]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
  s = (prec ? toFixedFix(n, prec) : "" + Math.round(n)).split('.');
```

File: public/js/extra/chart-pie.js [hunk 16]
```diff
@@ -0,0 +1,68 @@
+// Set new default font family and font color to mimic Bootstrap's default styling
+Chart.defaults.global.defaultFontFamily = 'Nunito', '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif';
```
Comment (Mamoon97150) [inline public/js/extra/chart-pie.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
Chart.defaults.global.defaultFontFamily = "Nunito", '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif';
```

File: public/js/sb-admin-2.js [hunk 17]
```diff
@@ -0,0 +1,56 @@
+(function($) {
+  "use strict"; // Start of use strict
+
+  // Toggle the side navigation
+  $("#sidebarToggle, #sidebarToggleTop").on("click", function(e) {
+    $("body").toggleClass("sidebar-toggled");
+    $(".sidebar").toggleClass("toggled");
+    if ($(".sidebar").hasClass("toggled")) {
+      $('.sidebar .collapse').collapse("hide");
```
Comment (Mamoon97150) [inline public/js/sb-admin-2.js:9]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
      $(".sidebar .collapse").collapse("hide");
```

File: public/js/extra/chart-area.js [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+  // *     return: '1 234,56'
+  number = (number + '').replace(',', '').replace(' ', '');
+  var n = !isFinite(+number) ? 0 : +number,
+    prec = !isFinite(+decimals) ? 0 : Math.abs(decimals),
+    sep = (typeof thousands_sep === 'undefined') ? ',' : thousands_sep,
+    dec = (typeof dec_point === 'undefined') ? '.' : dec_point,
+    s = '',
+    toFixedFix = function(n, prec) {
+      var k = Math.pow(10, prec);
+      return '' + Math.round(n * k) / k;
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:16]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
      return "" + Math.round(n * k) / k;
```

File: public/js/extra/chart-area.js [hunk 19]
```diff
@@ -0,0 +1,143 @@
...
+            borderWidth: 1,
+            xPadding: 15,
+            yPadding: 15,
+            displayColors: false,
+            intersect: false,
+            mode: "index",
+            caretPadding: 10,
+            callbacks: {
+              label: function(tooltipItem, chart) {
+                var datasetLabel = chart.datasets[tooltipItem.datasetIndex].label || '';
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:131]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
                var datasetLabel = chart.datasets[tooltipItem.datasetIndex].label || "";
```

File: public/js/extra/chart-area.js [hunk 20]
```diff
@@ -0,0 +1,143 @@
+// Set new default font family and font color to mimic Bootstrap's default styling
+Chart.defaults.global.defaultFontFamily = "Nunito", '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif';
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
Chart.defaults.global.defaultFontFamily = "Nunito", "-apple-system,system-ui,BlinkMacSystemFont,\"Segoe UI\",Roboto,\"Helvetica Neue\",Arial,sans-serif";
```

File: public/js/main.js [hunk 21]
```diff
@@ -156,15 +156,15 @@
...
   if (skilsContent) {
     new Waypoint({
       element: skilsContent,
       offset: '80%',
       handler: function(direction) {
-        let progress = select('.progress .progress-bar', true);
+        let progress = select(".progress .progress-bar", true);
         progress.forEach((el) => {
-          el.style.width = el.getAttribute('aria-valuenow') + '%'
+          el.style.width = el.getAttribute("aria-valuenow") + '%'
```
Comment (Mamoon97150) [inline public/js/main.js:167]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
          el.style.width = el.getAttribute("aria-valuenow") + "%"
```

File: public/js/extra/chart-area.js [hunk 22]
```diff
@@ -0,0 +1,143 @@
...
+      return '' + Math.round(n * k) / k;
+    };
+  // Fix for IE parseFloat(0.55).toFixed(0) = 0;
+  s = (prec ? toFixedFix(n, prec) : '' + Math.round(n)).split('.');
+  if (s[0].length > 3) {
+    s[0] = s[0].replace(/\B(?=(?:\d{3})+(?!\d))/g, sep);
+  }
+  if ((s[1] || '').length < prec) {
+    s[1] = s[1] || '';
+    s[1] += new Array(prec - s[1].length + 1).join('0');
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:25]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
    s[1] += new Array(prec - s[1].length + 1).join("0");
```

File: public/js/extra/chart-area.js [hunk 23]
```diff
@@ -0,0 +1,146 @@
...
+
+function numberFormat(number, decimals, decPoint, thousandsSep) {
+  // *     example: number_format(1234.56, 2, ',', ' ');
+  // *     return: '1 234,56'
+  number = (number + '').replace(',', '').replace(' ', '');
+  var n = !isFinite(+number) ? 0 : +number,
+    prec = !isFinite(+decimals) ? 0 : Math.abs(decimals),
+    sep = (typeof thousandsSep === "undefined") ? ',' : thousandsSep,
+    dec = (typeof decPoint === "undefined") ? '.' : decPoint,
+    s = '',
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:13]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
    s = "",
```

File: public/js/extra/chart-area.js [hunk 24]
```diff
@@ -0,0 +1,146 @@
+// Set new default font family and font color to mimic Bootstrap's default styling
+Chart.defaults.global.defaultFontFamily = "Nunito", "-apple-system,system-ui,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,sans-serif";
+Chart.defaults.global.defaultFontColor = '#858796';
+
+function numberFormat(number, decimals, decPoint, thousandsSep) {
+  // *     example: number_format(1234.56, 2, ',', ' ');
+  // *     return: '1 234,56'
+  number = (number + '').replace(',', '').replace(' ', '');
```
Comment (Mamoon97150) [inline public/js/extra/chart-area.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Mamoon97150/Blog-PHP/pullRequest?prid=7234961)
```suggestion
  number = (number + "").replace(',', '').replace(' ', '');
```

**Review by PixeleeCode — APPROVED** [review]

**Comment by PixeleeCode** [comment]
Pense à revoir le README

**Pull request merged**